### PR TITLE
Clarify missing classes and properties in the EditorInspector

### LINKED ISSUE #78469: Godot 4.0.3 - World Environment > Properties > Resources not displaying
### Godot version

Godot 4.0.3 - Windows - .NET - Standard (x86_64) 

### System information

Windows 11

### Issue description

I am getting an empty resources box when creating a new Environment. Its also happening on the World Environment > Camera Attributes resources.

I have tried Godot 4.0.3 MacOS Silicon version and can confirm that this issue does not occur. So this seems to only occur for Godot 4.0.3 Windows version.

I have attached a screenshot for reference.

Screenshot 2023-06-19 205423

### Steps to reproduce

1.) Create new Godot 4.0.3 Windows version project.
2.) Add the following Node Elements
- Node3D
-- WorldEnvironment

3.) Click on WorldEnvironment, go to Inspector.
4.) Click on Environment Dropdown and select 'New Environment'. 
5.) Click on 'Environment' > Environment (in Dropdown). At this point, the Environment resource list should appear but instead its just blank.

### Minimal reproduction project

[tut-3d-platformer.zip](https://github.com/godotengine/godot/files/11799232/tut-3d-platformer.zip)

### PR DESCRIPTION
**Pull request by ghost**
Implemented a "Button + Dialog" in the `EditorInspector` that warn and show which features were disabled by the "Editor Feature Profile".

https://github.com/godotengine/godot/assets/11199582/122f1ff9-e868-4aeb-a9c3-7803024136f2

### CODE REVIEW

**Comment by aaronfranke** [comment]
The ability to hide things in the editor is useful in educational contexts where you want to keep things simplified. Showing a message may go against this goal, because it would cause those users to worry about what is supposed to be hidden away from them. Anyway, I'm curious as to how these things could be disabled accidentally in #78469, it seems like something that has to be done very intentionally.

**Comment by felaugmar** [comment]
> The ability to hide things in the editor is useful in educational contexts where you want to keep things simplified. Showing a message may go against this goal, because it would cause those users to worry about what is supposed to be hidden away from them

I totally agree with that, if approved, probably an "editor settings" could be introduced to configure if this will or not be used (defaulting to `true`).

> it seems like something that has to be done very intentionally.

Yes, I imagine it being useful in the two situations:

1. You've never messed with this editor capability before (like if someone adjusted it for you);
2. You've adjusted it in the very past and just forgot about it.

**Comment by felaugmar** [comment]
During the development of this I was out of ideas for the warn "button style", so I've just made it simple.

If going to be approved, I guess it would be nice to review it, probably a small button with just an icon feels better.


During the development of this, I ran out of ideas for the "button style", so I kept it plain and simple.

If it's approved, I think it would be interesting to review and test something different, probably just a small icon would look better.

File: editor/editor_inspector.cpp [hunk 1]
```diff
@@ -2581,6 +2726,28 @@ void EditorInspector::_parse_added_editors(VBoxContainer *current_vbox, EditorIn
...
+	StringName class_name = object->get_class();
+
+	if (profile->is_class_disabled(class_name)) {
+		List<StringName> feature_chain;
+
+		feature_chain.push_back(class_name);
+
+		disabled_features_dialog->add_disabled_feature(feature_chain);
+
+		//won't see properties of a disabled class
```
Comment (AThousandShips) [inline editor/editor_inspector.cpp:2744]:
```suggestion
		// Won't see properties of a disabled class.
```
[Comment style](https://docs.godotengine.org/en/latest/contributing/development/code_style_guidelines.html#comment-style-guide) nitpick

File: editor/editor_inspector.cpp [hunk 2]
```diff
@@ -2696,638 +2867,656 @@ void EditorInspector::update_tree() {
...
 				}
-			}
 
-			if (category->icon.is_null() && !type.is_empty()) {
-				category->icon = EditorNode::get_singleton()->get_class_icon(type, "Object");
-			}
+				// Create an EditorInspectorCategory and add it to the inspector.
+				EditorInspectorCategory *category = memnew(EditorInspectorCategory);
+				main_vbox->add_child(category);
+				category_vbox = nullptr; //reset
```
Comment (AThousandShips) [inline editor/editor_inspector.cpp:2949]:
```suggestion
				category_vbox = nullptr; // Reset.
```

File: editor/editor_inspector.cpp [hunk 3]
```diff
@@ -2696,638 +2867,656 @@ void EditorInspector::update_tree() {
...
-					}
-					if (!editors[i].label.is_empty()) {
-						ep->set_label(editors[i].label);
-					} else {
-						// Use the existing one.
-						ep->set_label(property_label_string);
-					}
+					if (properties.size()) {
+						if (properties.size() == 1) {
+							//since it's one, associate:
```
Comment (AThousandShips) [inline editor/editor_inspector.cpp:3438]:
```suggestion
							// Since it's one, associate:
```

File: editor/editor_inspector.cpp [hunk 4]
```diff
@@ -2696,638 +2867,656 @@ void EditorInspector::update_tree() {
...
+							ep->property = properties[0];
+							ep->property_path = property_prefix + properties[0];
+							ep->property_usage = p.usage;
+							//and set label?
```
Comment (AThousandShips) [inline editor/editor_inspector.cpp:3442]:
```suggestion
							// And set label?
```

File: editor/editor_inspector.cpp (hunk 2 above)
Comment (felaugmar) [reply editor/editor_inspector.cpp:2949]: This is out of the scope of my changes, I just placed an `if` enclosing it, it's only "white space changes".

File: editor/editor_inspector.cpp (hunk 3 above)
Comment (felaugmar) [reply editor/editor_inspector.cpp:3438]: This is out of the scope of my changes, I just placed an `if` enclosing it, it's only "white space changes".

File: editor/editor_inspector.cpp (hunk 4 above)
Comment (felaugmar) [reply editor/editor_inspector.cpp:3442]: This is out of the scope of my changes, I just placed an `if` enclosing it, it's only "white space changes".

File: editor/editor_inspector.cpp (hunk 2 above)
Comment (AThousandShips) [reply editor/editor_inspector.cpp:2949]: The code is updated so it's best to get it in line, and not much work, so please update it anyway :)

**Comment by timothyqiu** [comment]
It'll probably be better if a warning is only shown when _all_ properties are hidden.

Missing some of the properties makes you think they're somehow hidden. But missing all of them makes you think it's a bug.

**Comment by felaugmar** [comment]
> It'll probably be better if a warning is only shown when _all_ properties are hidden.
> Missing some of the properties makes you think they're somehow hidden. But missing all of them makes you think it's a bug.

This may be true from the perspective of an experienced user, but probably not someone new to Godot.

New users blame themselves or blame the software, they don't ask whys 😅

**Comment by felaugmar** [comment]
This feature was not well received.

For future reference, solves issues like #78469.

**Pull request closed without merging**